_studio/mfx_lib/encode_hw/hevc/agnostic/g9/hevcehw_g9_roi.cpp

```diff
@@ -107,6 +107,10 @@ mfxStatus ROI::CheckAndFixROI(
 
     auto IsInvalidRect = [](const ROI::RectData& rect)
     {
+        if (rect.Top == 0 && rect.Left == 0 && rect.Right == 0 && rect.Bottom == 0)
+        {
+            return false;
+        }
         return ((rect.Left >= rect.Right) || (rect.Top >= rect.Bottom));
     };
```

Comment:
After CheckAndFixRect above we have here top/left aligned down and right/bottom aligned up for the cases when they are not aligned to BlockSize and right/bottom > top/left.
If top/left == right/bottom or top/left > right/bottom inside the same Block we would have top/left==right/bottom here and if top/left > right/bottom and they don't belong to the same Block we would have top/left > right/bottom.
Let me know if you don't agree with it?
So we have to consider a Rect as invalid only the way it was done before i.e. ((rect.Left >= rect.Right) || (rect.Top >= rect.Bottom))